controllers/osbuild_controller.go

```diff
@@ -47,16 +82,305 @@ type OSBuildReconciler struct {
 func (r *OSBuildReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx).WithValues("osbuild", req.Name)
+
+	osBuild, err := r.OSBuildRepository.Read(ctx, req.Name, req.Namespace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			logger.Error(err, fmt.Sprintf("OSBuild %s wasn't found", req.Name))
+			return ctrl.Result{}, err
+		}
+		logger.Error(err, fmt.Sprintf("OSBuild %s cannot be retrieve", req.Name))
+		return ctrl.Result{Requeue: true, RequeueAfter: requeueForShortDuration}, nil
+	}
+
+	if osBuild.DeletionTimestamp != nil {
+		// The OSBuild CRs that were created by that OSBuildConfig would be deleted
+		// thanks to setting controller reference for each OSBuild CR
+		return ctrl.Result{}, nil
+	}
+
+	lastBuildStatus := osbuildv1alpha1.OSBuildConditionType("")
+	if osBuild.Status.Conditions != nil {
+		conditionLen := len(osBuild.Status.Conditions)
+		lastBuildStatus = osBuild.Status.Conditions[conditionLen-1].Type
+	}
+
+	if osBuild.Status.ContainerComposeId == emptyComposeID {
+		// if the edge container wasn't created yet - schedule a new build
+		logger.Info("create an edge-container")
+		err = r.postComposeEdgeContainer(ctx, logger, osBuild)
+		if err != nil {
+			logger.Error(err, "failed to create an edge-container")
+			return ctrl.Result{Requeue: true, RequeueAfter: requeueForLongDuration}, nil
+		}
+
+		logger.Info("new job created for edge-container, requeue to sample its status")
+		return ctrl.Result{Requeue: true, RequeueAfter: requeueForLongDuration}, nil
+	}
+
+	if lastBuildStatus == startedContainerBuild {
+		// if the edge container already created but wasn't finish yet - check the build status
+		logger.Info("update the edge-container's compose ID job status")
+		composeStatus, err := r.updateContainerComposeStatus(ctx, logger, osBuild)
+		if err != nil {
+			logger.Error(err, "failed to get compose ID status")
+			return ctrl.Result{Requeue: true, RequeueAfter: requeueForShortDuration}, nil
+		}
+
+		// the build is still in progress - requeue
+		if composeStatus == composer.ComposeStatusValuePending {
+			logger.Info(fmt.Sprintf("the job ID %s, is still in progress", osBuild.Status.ContainerComposeId))
+			return ctrl.Result{Requeue: true, RequeueAfter: requeueForLongDuration}, nil
+		}
+
+		// the build was failed - return with error
+		if composeStatus == composer.ComposeStatusValueFailure {
+			err = fmt.Errorf("failed to build edge container")
+			logger.Error(err, "")
+			return ctrl.Result{}, err
+		}
 
-	// TODO(user): your logic here
+		// if the build was finished successfully and the target image is edge-container than return
+		if composeStatus == composer.ComposeStatusValueSuccess && osBuild.Spec.Details.TargetImage.TargetImageType == edgeContainerImgType {
+			logger.Info(fmt.Sprintf("the job ID %s, Finished", osBuild.Status.ContainerComposeId))
+			return ctrl.Result{}, nil
+		}
+	}
+
```

Comment:
I think we're missing a check here. When building container is done (successfully or not), you'll get another reconcile call. This time lastBuildStatus will not be `startedContainerBuild` but either `containerBuildDone` or `failedContainerBuild` (depending on success of failure of the build). 
Then, if it failed, or succeeded, but the target is Container, you need to complete the process.
I think you'll get the cleanest code if you always requeue at the end of the previous `if` statement. This way, once the build is done, the Reconcile method we'll be called again, the status will be fetched again.